analysis.md

```diff
@@ -0,0 +1,77 @@
+# Malware Analysis
+
+## Extracting the Source Code
+The file hash of the malicious document is `340795d1f2c2bdab1f2382188a7b5c838e0a79d3f059d2db9eb274b0205f6981`. I used [oletools](https://github.com/decalage2/oletools) to extract the VBA source code from the macro.
+
+## Layer 1
+The author of the script uses a lot of newline spacing to spread apart its contents. There is a function `ParsingA()` that appears to download content from the Internet, but it never seems to be used. You'll also notice that the author uses gratuitous string concatenation to confuse readers. For instance...
+
+```
+strComputer = "."
+Set objWMIService = GetObject("w" & "" & "in" & "" & "mgm" & "" & "ts" & "" & ":" & "" & "\\" & strComputer & "\r" & "" & "oot\c" & "" & "imv" & "" & "2")
+```
+...evaluates to the same thing as this: `Set objWMIService = GetObject("winmgmts:\\.\root\cimv2")`
+
+Also notice the large Base64 blob that `lStr` contains. The script decodes this blob and decompresses it for execution.
+
+### Unraveling the Base64
+Below is the Python3 script I use to decode the Base64 blob mentioned earlier.
+
+```
+import base64
+import sys
+
+ifile = open(sys.argv[1], "r")
+ofile = open(sys.argv[2], "w")
```

Comment:
A fairly linux standard thing to do here is to write results to stdout rather than a file. That way, if the user of the script wants to put it to a file, they can `decoder.py encoded.txt > decoded.txt`